uhi/typing/plottable.py

```diff
@@ -46,10 +59,20 @@ def __eq__(self, other: Any) -> bool:
         """
 
 
+PlottableAxisContinuous = PlottableAxisGeneric[Tuple[float, float]]
+PlottableAxisInt = PlottableAxisGeneric[int]
+PlottableAxisStr = PlottableAxisGeneric[str]
+
+PlottableAxis = Union[PlottableAxisContinuous, PlottableAxisInt, PlottableAxisStr]
+
+
 @runtime_checkable
 class PlottableHistogram(Protocol):
     axes: Sequence[PlottableAxis]
 
+    weighted: bool
```

Comment:
I don't see why `weighted` has to be exposed. Whether the histogram is weighted or not should be an implementation detail, the plotter simply requests values and variances. It does not need to know whether the content is weighted.

Reply:
You need something that can inform a smart plotter that "variances" are useful new information or just values again. A dumb plotter can just always plot variances, but a smarter one should be able to avoid drawing error bars if you are not using a weighted storage. I would want a plotter to add error bars if I use a Weight() storage, but not if I use a Double() storage, it would be plotting incorrect information if I filled with a weight. Maybe the name is not ideal? `has_variance` or something like that?

Basically, this should not plot error bars by default, as they are wrong:

```python
h = Hist(..., storage=Double())
h.fill(values, weight=weights)
mplhep.histplot(h)
```

Changing `Double()` to `Weight()` should then add the error bars, or maybe explicitly forcing them.

Reply:
That's why this was Optional, but we moved to just providing some way for a smart plotting function to tell if this is really new information or not.

Reply:
You always want error bars, even for normal histograms where variance = value.

Reply:
`has_variance` makes more sense then, but it is still weird to have variance() always return an array. If that is optional, it should be None if there is no variance information and an array otherwise. I still think that `has_variance` should not be there. Python has a natural way to return optional results (return the result or None).

Reply:
Maybe the most important thing was fixing `.counts` to always return, and we could let `.variances` be Optional. @jpivarski?